### PR TITLE
Feature/aoaf 288 generate the presenter

### PR DESCRIPTION
**Pull request by myadavch**
the following changes are related to generating presenterId through oracle db sequence.

### CODE REVIEW

File: src/main/java/uk/gov/companieshouse/presenter/account/processor/PresenterAccountMessageProcessor.java [hunk 1]
```diff
@@ -41,7 +45,20 @@ public PresenterAccountDetailsApi processPresenterCreatedMessage(Message<Present
...
+
         logger.debug("Presenter details fetched successfully from presenter-account-api for " + presenterAccountDetailsApi.getPresenterDetailsId());
         return presenterAccountDetailsApi;
     }
+
+    /**
+     * This method will generate the presenterid.
+     * @return
+     */
+    private String generatePresenterId(){
```
Comment (amartin7663) [inline src/main/java/uk/gov/companieshouse/presenter/account/processor/PresenterAccountMessageProcessor.java:59]: Could do with a dedicated unit test to test this function

File: src/main/java/uk/gov/companieshouse/presenter/account/processor/PresenterAccountMessageProcessor.java [hunk 2]
```diff
@@ -41,7 +45,20 @@ public PresenterAccountDetailsApi processPresenterCreatedMessage(Message<Present
...
+        String presenterId = "E"  + String.format("%010d", presenterAccountDetailsRepository.getPresenterDetailsId());
+        logger.debug(" Presenter Id from sequence: " + presenterId);
```
Comment (amartin7663) [inline src/main/java/uk/gov/companieshouse/presenter/account/processor/PresenterAccountMessageProcessor.java:61]: Space a the front of this debug message

File: src/main/java/uk/gov/companieshouse/presenter/account/repository/PresenterAccountDetailsRepository.java [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+
+import org.springframework.data.jdbc.repository.query.Query;
+import org.springframework.data.repository.CrudRepository;
+import org.springframework.stereotype.Repository;
+import uk.gov.companieshouse.api.model.presenteraccount.PresenterAccountDetailsApi;
+
+@Repository
+public interface PresenterAccountDetailsRepository extends CrudRepository<PresenterAccountDetailsApi, Long> {
+
+    static final String  SEQUENCE_NAME = "CHS_PRESENTER_ID_SEQ";
```
Comment (amartin7663) [inline src/main/java/uk/gov/companieshouse/presenter/account/repository/PresenterAccountDetailsRepository.java:11]: Double space in here, also can this be private

**Comment by ch-code-analysis** [comment]
CI: Security warnings found!

File: src/main/java/uk/gov/companieshouse/presenter/account/processor/PresenterAccountMessageProcessor.java (hunk 1 above)
Comment (myadavch) [reply src/main/java/uk/gov/companieshouse/presenter/account/processor/PresenterAccountMessageProcessor.java:57]: added the happy path scenario. unhappy path already covered under retryable and non retryable exception.

**Comment by ch-code-analysis** [comment]
CI: Security warnings found!

File: src/main/java/uk/gov/companieshouse/presenter/account/processor/PresenterAccountMessageProcessor.java (hunk 2 above)
Comment (myadavch) [reply src/main/java/uk/gov/companieshouse/presenter/account/processor/PresenterAccountMessageProcessor.java]: fixed

File: src/main/java/uk/gov/companieshouse/presenter/account/repository/PresenterAccountDetailsRepository.java (hunk 3 above)
Comment (myadavch) [reply src/main/java/uk/gov/companieshouse/presenter/account/repository/PresenterAccountDetailsRepository.java]: fixed

File: src/main/java/uk/gov/companieshouse/presenter/account/repository/PresenterAccountDetailsRepository.java [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+
+    @Query (value="select " + SEQUENCE_NAME + ".NEXTVAL FROM DUAL")
```
[the remark this answers is not in the document]
Comment (myadavch) [reply src/main/java/uk/gov/companieshouse/presenter/account/repository/PresenterAccountDetailsRepository.java]: fixed

File: src/main/java/uk/gov/companieshouse/presenter/account/repository/PresenterAccountDetailsRepository.java [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+    public Long  getPresenterDetailsId();
```
[the remark this answers is not in the document]
Comment (myadavch) [reply src/main/java/uk/gov/companieshouse/presenter/account/repository/PresenterAccountDetailsRepository.java]: fixed

File: src/main/java/uk/gov/companieshouse/presenter/account/processor/PresenterAccountMessageProcessor.java [hunk 6]
```diff
@@ -37,11 +41,22 @@ private PresenterAccountDetailsApi callPresenterAccountApi(String presenterId) {
     public PresenterAccountDetailsApi processPresenterCreatedMessage(Message<PresenterCreated> presenterCreatedMessage) {
         final PresenterCreated presenterCreatedPayload = presenterCreatedMessage.getPayload();
         String presenterCreatedID = presenterCreatedPayload.getId();
-        if(presenterCreatedID == null || presenterCreatedID.trim().length() == 0){
+        if(presenterCreatedID == null || presenterCreatedID.trim().isEmpty()){
             throw new NonRetryableErrorException("Presenter Id in the kafka message is blank");
         }
         PresenterAccountDetailsApi presenterAccountDetailsApi = callPresenterAccountApi(presenterCreatedID);
+        presenterAccountDetailsApi.setPresenterDetailsId(generatePresenterId());
```
Comment (amartin7663) [inline src/main/java/uk/gov/companieshouse/presenter/account/processor/PresenterAccountMessageProcessor.java:48]: The presenter id and the presenter details id are two different values, the details id is the id of the mongo record.

**Comment by ch-code-analysis** [comment]
CI: Security warnings found!

File: src/main/java/uk/gov/companieshouse/presenter/account/processor/PresenterAccountMessageProcessor.java (hunk 6 above)
Comment (myadavch) [reply src/main/java/uk/gov/companieshouse/presenter/account/processor/PresenterAccountMessageProcessor.java]: fix the mistake.

**Comment by ch-code-analysis** [comment]
CI: Security warnings found!

**Comment by ch-code-analysis** [comment]
CI: Security warnings found!

**Review by amartin7663 — APPROVED** [review]

**Comment by ch-code-analysis** [comment]
CI: Security warnings found!

**Comment by ch-code-analysis** [comment]
CI: Security warnings found!

**Pull request merged**